core/tx_pool.go

```diff
@@ -387,11 +387,26 @@ func (pool *TxPool) reset(oldHead, newHead *types.Header) {
 		} else {
 			// Reorg seems shallow enough to pull in all transactions into memory
 			var discarded, included types.Transactions
-
 			var (
 				rem = pool.chain.GetBlock(oldHead.Hash(), oldHead.Number.Uint64())
 				add = pool.chain.GetBlock(newHead.Hash(), newHead.Number.Uint64())
 			)
+			if rem == nil {
```

Comment:
There's no possible case where `add == nil`? Wondering if it's worth having an additional sanity check here to avoid a panic in this case.

I don't have a lot of experience with the underlying `GetBlock()` function, but it looks like it's possible and allowed for it to return a nil pointer.

Reply:
If there is, that's definitely an error. What happens here is that we just got a signal that the chain reorged, and it tells us the old head and the new head. 
When we ask for more info about the old head, it can be nil if the underlying cause was 'setHead'. However, if the chain can't give us info aobut the new head, then something is very wrong.

Reply:
Ok got it, thanks for the context!